### ISSUE TITLE
Request: Add support for eslint v9

### ISSUE THREAD

**Issue by joshunrau** [azat-io/eslint-plugin-perfectionist#154]
### Describe the bug

Currently, eslint >= 8.0.0 is listed as a peer dependency. I can use flat config with v8 and the plugin (example below). It should be confirmed everything works with v9, after which is can be added to peers.  

### Code example

```js
import perfectionist from 'eslint-plugin-perfectionist';

export default [
  {
    files: ['**/*.js'],
    plugins: {
      perfectionist
    },
    rules: {
      ...perfectionist.configs['recommended-natural'].rules
    }
  }
];

```

### ESLint version

v8.57.0

### ESLint Plugin Perfectionist version

v2.10.0

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by pauliesnug** [comment]
Also there are no included TypeScript declaration files, which causes issues with Flat Config type checking. I think this is in scope of #90 as well.

**Comment by azat-io** [comment]
Yes, we are currently working on implementing this feature.

**Comment by what1s1ove** [comment]
> Yes, we are currently working on implementing this feature.

Hey @azat-io ! 

Could you please let me know if you are planning to add a `d.ts` file for this plugin? Currently, there is an error because it is missing.

<img width="1044" alt="image" src="https://github.com/azat-io/eslint-plugin-perfectionist/assets/42296182/355f0fd9-27da-4021-b7b8-65953d7b6c01">

**Comment by azat-io** [comment]
Yes, we're planning to add ts support!

**Comment by azat-io** [comment]
Thank you for your issue!

We've updated the way configs are exported to match ESLint standards in 2a7eec20fc8188a01418690577f2888df8fe1ff1.

Also added support for types in 2b63ab558f431a33fbdfb103c3279ec34011d135.

All of this was released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054

**Comment by pauliesnug** [comment]
@azat-io This is amazing! Thank you for your work <3
